internal/services/cdn/cdn_frontdoor_rule_resource.go

```diff
@@ -618,12 +618,12 @@ func resourceCdnFrontDoorRuleCreate(d *pluginsdk.ResourceData, meta interface{})
 	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	ruleSetId, err := parse.FrontDoorRuleSetID(d.Get("cdn_frontdoor_rule_set_id").(string))
+	ruleSet, err := parse.FrontDoorRuleSetIDInsensitively(d.Get("cdn_frontdoor_rule_set_id").(string))
```

Comment:
this is coming from the config/is validated so we can parse this case-sensitively:

```suggestion
	ruleSet, err := parse.FrontDoorRuleSetID(d.Get("cdn_frontdoor_rule_set_id").(string))
```

Reply:
Fixed.